### DISCUSSION TITLE
it is possible to add e621 support in the future?

### DISCUSSION THREAD

**Discussion by Tihf** [trickerer01/Ruxx#15] [Ideas]
https://e621.net/
seems similar to rule34 and have some arts that are not in rule34, would be a cool addition if not too complicated.

**Comment by trickerer01** [comment]
I'll consider it.
I had a gigantic headache trying to implement RZ since their website engine is so different from any booru, so I actually went and made myself a checklist of what I have to check before even thinking about trying to support another website.

They seem to build their API based on danbooru engine, that's good. Pools support, post relationships - all that is intersting to implement to finally make some features supported by more than one module.
On the other hand the dev team seems very cautious about people abusing their API through downloaders but don't specify the limits, I don't really want people get blocked because of that or because I made Ruxx impersonate a real browser, well that's probably not going to be an issue unless you actually override download threads limit.

**Comment by trickerer01** [comment]
Well I trudged through their forums and found some more info on API limitations. And no, I'm not going to try to circumvent any of those. Max 40 tags per search, only one wildcard tag is allowed, ok - some more negative tags extraction code. One API request per second, except for actual downloads. Some custom user-agent header is required, I'll have to force it over anything provided by the user to comply. Their documentation says 'or' groups aren't entirely reliable, also syntax is completely different from RX one, oh well - all the 'or' groups will be unwrapped just like with most other modules, thankfully the code is already there.
The website is actively supported by the staff and they even do db exports, on the other hand their API interface may change at some point and I won't know it since tests certainly won't cover everything.

Actually all mentioned work is already completed and working, hopefully will be pushed soon. Just making double sure I didn't miss anything, adding some additional features along the way. Also have to add wrapper support: https://github.com/trickerer01/download-multi-async-wrapper

Now I have a question since I'm curious: why have you suggested e621.net exactly? Apparently this website is pretty popular and there is half a dozen downloaders exist supporting it, not counting dedicated browser plugins, a couple downloaders are dedicated, too. I'm curious since the last time I got a feature request it was for ML #12 and I'm not sure what to think.

**Reply by Tihf** [reply]
mostly because your downloader was the easiest i found after some search, but with those dedicated downloaders you mentioned, if they are as easy to use as your own, you probably should not even bother adding e621 support tbh

**Comment by trickerer01** [comment]
https://github.com/trickerer01/Ruxx/commit/96b86fda6e9c8b21fb7cb3058a814f5573706b1a

**Comment by trickerer01** [comment]
https://github.com/trickerer01/Ruxx/releases/tag/1.6.673
